src/main/java/org/spongepowered/api/util/command/MessageSink.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util.command;
+
+import org.spongepowered.api.text.Text;
+
+/**
+ * Represents a function that takes a message and distributes it to the appropriate targets.
+ */
+public interface MessageSink {
+    /**
+     * Handle transforming the input message appropriately.
+     *
+     * @param text The message to send
+     * @return The transformed text. May be input.
+     */
+    Text transformMessage(Text text);
```

Comment:
So if I needed to prepend a rank?

On Thu, May 21, 2015 at 5:41 PM zml <notifications@github.com> wrote:

> In src/main/java/org/spongepowered/api/util/command/MessageSink.java
> <https://github.com/SpongePowered/SpongeAPI/pull/694#discussion_r30855386>
> :
>
> > + */
> > +package org.spongepowered.api.util.command;
> > +
> > +import org.spongepowered.api.text.Text;
> > +
> > +/**
> > + * Represents a function that takes a message and distributes it to the appropriate targets.
> > + */
> > +public interface MessageSink {
> > +    /**
> > +     * Handle transforming the input message appropriately.
> > +     *
> > +     * @param text The message to send
> > +     * @return The transformed text. May be input.
> > +     */
> > +    Text transformMessage(Text text);
>
> This method is intended to produce sink-specific output for a given input
> -- so in a channels system, this would be the method that prepends the name
> of the channel. No information is really known about the type of message,
> and that is intentional to allow any message to pass through channels
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpongePowered/SpongeAPI/pull/694/files#r30855386>.
>